### ISSUE TITLE
Wrong value in P1meter

### ISSUE THREAD

**Issue by uffer** [domoticz/domoticz#4874]
Like in:  Overflow at P1 smart meter #4521 i have the same isue with my p1meter.

It says the value of delivery 18446744073709552.000

I'm running the latest stable and also with SQLite3 I can't find the value in the DB.

sqlite3 domoticz.repair | grep '18446744073709552.000' doesn't return a value and I can't find the value in the SQLite browser either.

Do you know where I can reset the value to 0. I don't use Node-Red by the way.

![afbeelding](https://user-images.githubusercontent.com/18116064/125187607-b94beb00-e230-11eb-842a-07991df19d0c.png)

Running on Ubuntu btw.

![afbeelding](https://user-images.githubusercontent.com/18116064/125188963-3168df80-e236-11eb-8ba2-d795ce3b34d0.png)

**Comment by Olifant1990** [comment]
1. What kind of device do you use? and what protocol does it use? z-wave/ethernet/wifi etc.
2. No, that "Levering" is a total, summed up for today, so you won't find that specific value in the database.

I see you are using 202.2 build 13016. Maybe you can try to update to the latest beta.
There are done a lot of fixes...

**Comment by uffer** [comment]
1. AM550-ED1 CDMA meter -> P1-USB cable (like this: https://www.sossolutions.nl/slimme-meter-kabel) ESXI (7.0) -> Ubuntu 20.04.2) and monitoring this since 2015 (Meter, ESXI host and ubuntu are upgraded during this time)
2. Ok, when I check the smartmeter itself it is saying 0.0 for today (don't have PV on my roof) and this value is there since a couple of weeks. Is there any way to clear this value?

**Comment by gizmocuz** [comment]
Maybe you had a fuse blown or another hickup ... you can clear the value in the database
Export the database from the settings page
Edit the database with (for example) SQLite Expert Personal Edition
Go to the devicetable , find your p1 meter, and there you should see this large value in the 'svalue' field
Next import the database into domoticz again via the settings page
If this does not work, then it seems you receive this value from the smartmeter

**Comment by uffer** [comment]
I assume this is the MultiMeter or the MultiMeter_Calander, but it doesn't contain the value. Also a:

`~# sqlite3 domoticz.repair | grep 'CounterDelivToday'`

or a:

`~# sqlite3 domoticz.repair | grep '18446744073709552.000'`

on the CLI returns nothing.

The strange thing is that the value is there every day and that it is nowhere to be found. Can it be reset somewhere (perhaps in the DB)?

![afbeelding](https://user-images.githubusercontent.com/18116064/125335946-6caa0200-e34d-11eb-824b-5a818640f712.png)
![afbeelding](https://user-images.githubusercontent.com/18116064/125336013-821f2c00-e34d-11eb-8b7b-14f8c054b7a1.png)

As you can see I've been running this database stable since 2015 and this is the first "glitch" I come across that I can't solve myself. Hope you have a solution :-)

---edit---
When I do a DB restore from May, the error was not there yet.
![afbeelding](https://user-images.githubusercontent.com/18116064/125337711-9b28dc80-e34f-11eb-8473-52eba11e079c.png)
![afbeelding](https://user-images.githubusercontent.com/18116064/125338258-43d73c00-e350-11eb-8459-5fe7f7af00c0.png)
![afbeelding](https://user-images.githubusercontent.com/18116064/125338456-83058d00-e350-11eb-95e8-de37e1539783.png)

And 5 minutes later there it is again....
![afbeelding](https://user-images.githubusercontent.com/18116064/125338714-d1b32700-e350-11eb-86fa-f2da4959d59b.png)

**Comment by gizmocuz** [comment]
No, I think you have to look in the Device Table, not the calendar tables...

**Comment by uffer** [comment]
OK, funny thing is that you can see the IDX 96 but the value is not present:
![afbeelding](https://user-images.githubusercontent.com/18116064/125670622-00d412f0-2081-45ca-8c31-64ec9f2f5b7c.png)

but if I do a: https://FQDN/json.htm?type=devices&rid=96

It is visible in the output:
![afbeelding](https://user-images.githubusercontent.com/18116064/125670979-d60ac105-010a-45fd-95d0-228bd5042274.png)

Would you perhaps know where the value: CounterDelivToday should be?

**Comment by Olifant1990** [comment]
I think you must have a very small or a very large number somewhere in your multimeter table?
positive or negative?

edit... but maybe it must have been shown in the graph too...
but can you look in that table?

**Comment by uffer** [comment]
Ok I think I found it:
![afbeelding](https://user-images.githubusercontent.com/18116064/126046923-6c67b86e-68bf-4987-8237-62cfd957c34e.png)

The only thing is when i try to delete the value with shift-mouseclick it wont be deleted.
![afbeelding](https://user-images.githubusercontent.com/18116064/126047034-04fd7b86-f2db-42f1-a48e-5f8359651bc9.png)

When i check the DB is counter 2 and 4 for delivery?
![afbeelding](https://user-images.githubusercontent.com/18116064/126047089-47577de3-4b18-4e46-a6be-4e65f9d193e2.png)

**Comment by uffer** [comment]
![afbeelding](https://user-images.githubusercontent.com/18116064/126493702-e4eaca35-9b8e-45e1-88ce-f436a552c597.png)

![afbeelding](https://user-images.githubusercontent.com/18116064/126494814-72b47eae-2e16-4276-8f7e-4f2a08b40778.png)

Could this be the problem?


---edit----
When I delete all the records wher in Multimeter where I can see weird value's in Value2 and in MultiMeter_Calander, Counter2 and import the DB after that it seems that most of the weird data is almost gone.

Now all the history is gone to :-(

Is there a way where I can only delete in multiple rows (like 2700) only the value's in the specific colums? So I can keep my history?

![afbeelding](https://user-images.githubusercontent.com/18116064/126502544-d617ba61-17ca-4747-b6ac-09def6bc8b89.png)

**Comment by Olifant1990** [comment]
Is your smart meter cable in Domoticz connected as hardware?
Or do you have your own sotware reading the P1 telegrams and sending it to domoticz?

@gizmocuz if first, could it be that Domoticz is not handling the CRC checksum of the telegrams?
I am personally reading my P1 data with an ESP8266 and I have seen a number of misreadings (read disturbings) in the serial readings. This all was over when I implemented the checksums and only using valid telegrams from that moment on.

**Comment by uffer** [comment]
Yes it is connected as hardware directly to Domoticz.

**Comment by Olifant1990** [comment]
Ok, in your hardware settings, is the option "CRC message validation" enabled? so, green?

**Comment by uffer** [comment]
Yes, it is:

![afbeelding](https://user-images.githubusercontent.com/18116064/126877944-3799ecef-2f8e-4488-bc32-c3cb40feb296.png)

**Comment by uffer** [comment]
Ok, I deleted a lot of strange negative value's in multimeter_calander and multimeter.:
![afbeelding](https://user-images.githubusercontent.com/18116064/131359177-5ed8600a-0aea-4648-be35-68d806dd5b5e.png)

(Thanx to @Knielen on https://gathering.tweakers.net/forum/list_message/68036538#68036538)

And deleted the value's in counter 2 and 4 to 0 in multimeter_calander and multimeter value 2 and 6:
![afbeelding](https://user-images.githubusercontent.com/18116064/131360397-ba6f73f8-0029-4778-adab-8003b2ee5d4f.png)

Now it show up like this:
![afbeelding](https://user-images.githubusercontent.com/18116064/131334086-4ebe056e-2aef-4184-9053-ee1600617650.png)

The only thing is that the graph still looks like this (and I dont have solarpanels (yet))
![afbeelding](https://user-images.githubusercontent.com/18116064/131360633-1137d737-f508-4c53-9a17-624b1e67f9d2.png)

Do you know @gizmocuz  and @Olifant1990 where I can find this value?

But the history is still present:

![afbeelding](https://user-images.githubusercontent.com/18116064/131393016-b9a4659a-a70c-46d5-99d2-c78e0e510041.png)

**Comment by uffer** [comment]
I think my database is corrupt.

Removed the P1 meter in Hardware, rebooted the machine and tried to add the P1 meter again:

![image](https://user-images.githubusercontent.com/18116064/135838933-e1f8bcce-9a7a-4bf0-8821-dfe04e6fef14.png)

But now no new values ​​are shown in the log. When I downloaded the database and opened it in SQLite, the data is still present in multimeter_calander and Multimeter.

I don't know if anyone recognizes this behavior and know what else I can do other than start over (which really sucks, there's 5 years of info in the DB).

**Comment by gizmocuz** [comment]
@uffer , If you have your baudrate set correctly, no rate limit  (=0), then you should receive values on the P1 port.
You can also open the serial port (when not in use!!) with for example putty (set the right baudrate) and see if you receive anything (might look malformed but you should see some data)
Also check the log level (under the P1 hardware) and enable Info/Status/Error

**Comment by uffer** [comment]
According to the following table on page: [https://www.netbeheernederland.nl/_upload/Files/Slimme_meter_15_a727fce1f1.pdf](url)

Would there have been a return of PV panels that I did not have....

`cu -l /dev/serial/by-id/usb-FTDI_FT232R_USB_UART_A5YXDA17-if00-port0 -s 115200 --parity=none -E q`

[1-3:0.2.8(50)]
0-0:1.0.0(211011113048S)
0-0:96.1.1(Equipment identifier)
1-0:1.8.1(005929.430*kWh)
1-0:1.8.2(006787.200*kWh)
1-0:2.8.1(000000.011*kWh)
1-0:2.8.2(000000.006*kWh)
0-0:96.14.0(0002)
1-0:1.7.0(00.150*kW)
1-0:2.7.0(00.000*kW)
0-0:96.7.21(00005)
0-0:96.7.9(00003)
1-0:99.97.0(1)(0-0:96.7.19)(171028010321S)(0000000349*s)
1-0:32.32.0(00010)
1-0:52.32.0(00011)
1-0:72.32.0(00011)
1-0:32.36.0(00001)
1-0:52.36.0(00001)
1-0:72.36.0(00001)
0-0:96.13.0()
1-0:32.7.0(232.2*V)
1-0:52.7.0(231.8*V)
1-0:72.7.0(230.6*V)
1-0:31.7.0(000*A)
1-0:51.7.0(000*A)
1-0:71.7.0(000*A)
1-0:21.7.0(00.034*kW)
1-0:41.7.0(00.059*kW)
1-0:61.7.0(00.058*kW)
1-0:22.7.0(00.000*kW)
1-0:42.7.0(00.000*kW)
1-0:62.7.0(00.000*kW)
0-1:24.1.0(003)
0-1:96.1.0(4730303332353635353737343939313138)
0-1:24.2.1(211011113011S)(00104.232*m3)

![afbeelding](https://user-images.githubusercontent.com/18116064/136771401-12876394-ff2e-42d5-8fa4-ef21a7853877.png)



So now I don't think it is an error in domoticz but in the smartmeter itself.

**Comment by Olifant1990** [comment]
I wouldn't bother those very small numbers.
They can theoretically happen with interference on the net.
But you can give a message to your provider if you want.

But... You seem te get data out of your meter. So, I don't think that one is faulty.
You should still check the same settings (baud-rate etc) again in Domoticz and enable debug like @gizmocuz said.

**Comment by Olifant1990** [comment]
More to say: Is that a full dump of the output?
What I am missing is a CRC checksum at the end.
If that is not send and interpreted by Domoticz, all of your strange values can happen by misreadings, not validated if your data was received correctly.

A telegram usally starts with a / and ends with a !.
After the ! you usually have a CRC of 4 characters.

See http://domoticx.com/p1-poort-slimme-meter-hardware/

**Comment by uffer** [comment]
Ah ok, so no worry's about the small numbers. But because of my not very good relationship with the network operator, I'll leave it at that.

The complete telegram looks like this:

1-3:0.2.8(50)
0-0:1.0.0(211011134103S)
0-0:96.1.1(4530303434303037303033313437373137)
1-0:1.8.1(005929.430*kWh)
1-0:1.8.2(006788.498*kWh)
1-0:2.8.1(000000.011*kWh)
1-0:2.8.2(000000.006*kWh)
0-0:96.14.0(0002)
1-0:1.7.0(00.603*kW)
1-0:2.7.0(00.000*kW)
0-0:96.7.21(00005)
0-0:96.7.9(00003)
1-0:99.97.0(1)(0-0:96.7.19)(171028010321S)(0000000349*s)
1-0:32.32.0(00010)
1-0:52.32.0(00011)
1-0:72.32.0(00011)
1-0:32.36.0(00001)
1-0:52.36.0(00001)
1-0:72.36.0(00001)
0-0:96.13.0()
1-0:32.7.0(231.5*V)
1-0:52.7.0(232.7*V)
1-0:72.7.0(231.8*V)
1-0:31.7.0(002*A)
1-0:51.7.0(000*A)
1-0:71.7.0(000*A)
1-0:21.7.0(00.486*kW)
1-0:41.7.0(00.060*kW)
1-0:61.7.0(00.056*kW)
1-0:22.7.0(00.000*kW)
1-0:42.7.0(00.000*kW)
1-0:62.7.0(00.000*kW)
0-1:24.1.0(003)
0-1:96.1.0(4730303332353635353737343939313138)
0-1:24.2.1(211011134011S)(00104.232*m3)
!05E0

For the P1 meter it is connected to domoticz like Gizmocuz said.

![afbeelding](https://user-images.githubusercontent.com/18116064/136784698-d6d376af-6903-46c5-b08f-f247b1199007.png)

**Comment by Olifant1990** [comment]
I am still missing the identifier part in your P1 telegram,
like you can find on http://domoticx.com/p1-poort-slimme-meter-hardware/
for example "/Ene5\T210-D ESMR5.0"
This is also critcal data which is calculated in the CRC.
If this is missing and Domoticz recalculates the CRC, it is never complete/correct, so no data.
Maybe you have to find the problem there then?

**Comment by Olifant1990** [comment]
Also, try for debug to disable CRC message validation for that hardware.
you then should receive values, but they are not validated then and can have misreading.

**Comment by uffer** [comment]
Oh sorry it is there:-)

/ISK5\2M550T-1012

1-3:0.2.8(50)
0-0:1.0.0(211011141824S)
0-0:96.1.1(4530303434303037303033313437373137)
1-0:1.8.1(005929.430*kWh)
1-0:1.8.2(006788.771*kWh)
1-0:2.8.1(000000.011*kWh)
1-0:2.8.2(000000.006*kWh)
0-0:96.14.0(0002)
1-0:1.7.0(00.233*kW)
1-0:2.7.0(00.000*kW)
0-0:96.7.21(00005)
0-0:96.7.9(00003)
1-0:99.97.0(1)(0-0:96.7.19)(171028010321S)(0000000349*s)
1-0:32.32.0(00010)
1-0:52.32.0(00011)
1-0:72.32.0(00011)
1-0:32.36.0(00001)
1-0:52.36.0(00001)
1-0:72.36.0(00001)
0-0:96.13.0()
1-0:32.7.0(237.8*V)
1-0:52.7.0(237.0*V)
1-0:72.7.0(236.0*V)
1-0:31.7.0(000*A)
1-0:51.7.0(000*A)
1-0:71.7.0(000*A)
1-0:21.7.0(00.111*kW)
1-0:41.7.0(00.062*kW)
1-0:61.7.0(00.060*kW)
1-0:22.7.0(00.000*kW)
1-0:42.7.0(00.000*kW)
1-0:62.7.0(00.000*kW)
0-1:24.1.0(003)
0-1:96.1.0(4730303332353635353737343939313138)
0-1:24.2.1(211011141507S)(00104.232*m3)
!707D

I have for the test the crc validation turned off.

**Comment by uffer** [comment]
Next day update:
![afbeelding](https://user-images.githubusercontent.com/18116064/136907927-0827707d-9c6d-40c8-b7ef-e76ea21e7639.png)

Unfortunately, CounterDelivToday's value remains at: 18446744073709552,000 kWh.

Either this is a field in the database (which I can't find) or it's a calculation of fields that might need to be set to 0.

I have re-enabled CRC validation.

**Comment by Olifant1990** [comment]
Well, based on the fact you GET DATA when CRC is DISABLED and NO DATA when CRC is ENABLED, my suspicion was right.
Or your data is corrupt, or the CRC checking method in Domoticz is not right.
First this needs to be fixed before looking for errors in the database/other calculations.
CounterDelivToday is, as I remember well, indeed a calculation, not a column in the database.

Maybe @gizmocuz or someone else can have a deeper look into this?

**Comment by gizmocuz** [comment]
@Olifant1990 , yes that is correct, it uses the last received meter value and the first value of the day

**Comment by uffer** [comment]
Ok, little bump, I did an upgrade to the latest beta with some hope for my rare counter_delivery problem. But no luck.

So I tried out what will happen if do an new installation with only the P1 meter in it. SO fresh ubuntu server install with only Domoticz (latest beta) on it.

But same problem.

![afbeelding](https://user-images.githubusercontent.com/18116064/151815591-334726e0-324d-4289-a8ab-88e36eb11d09.png)

![afbeelding](https://user-images.githubusercontent.com/18116064/151815926-bbd73219-6f9d-4de6-8f0c-a0ffd16eef43.png)

Could it be that the small delivery numbers are cousing this behavior?

**Comment by Olifant1990** [comment]
Is your CRC still disabled? Then Domoticz can insert "wrong" data

**Comment by uffer** [comment]
No, it was enabled (by default in the new domoticz installation).

**Comment by uffer** [comment]
Same problem but with a different person/Domoticz installation

https://gathering.tweakers.net/forum/list_message/70488604#70488604

I keep hoping it can be solved.

**Comment by fwieringen** [comment]
Experience the same issue with my P1 configuration. Started a [forum thread](https://www.domoticz.com/forum/viewtopic.php?t=37957) for this issue.

Somehow, my meter reported some delivery back to the grid on Feb 3. The Return Today value has been reporting 18446744073709552.000 as of that moment.

As @gizmocuz indicated, it's a calculated value. But I'm not sure what goes wrong with this calculation. When I look at the MultiMeter table for the first value of today, the Delivery counters are at 571 and 2. The current P1 result is also 571 and 2. 

Was looking for a dirty fix to make a database fix or have a script fix the database whenever the issue occurs, but I'm unable to determine where the calculation goes wrong...

**Comment by fwieringen** [comment]
Alright, did some investigation in the code myself. Had a look in the WebServer.cpp file which produces the content of the API calls. This contains the following code (line 10317+).

```
powerdeliv1 = (powerdeliv1 < 10) ? 0 : powerdeliv1;
powerdeliv2 = (powerdeliv2 < 10) ? 0 : powerdeliv2;
...
unsigned long long powerdeliv = powerdeliv1 + powerdeliv2;
```

This code reads the meter value and gets the current delivery total. However, since one of my counters is 2, this will be set  to 0 and therefore, my total delivery value is not 571+2 = 573, but 571+0.

This is then inserted in the API call:

```
root["result"][ii]["Counter"] = szTmp;
musage = double(powerdeliv) / EnergyDivider;
```

Later in that same file, the first values of the day are loaded from the database and the daily delivery is calculated:

```
result2 = m_sql.safe_query("SELECT MIN(Value1), MIN(Value2), MIN(Value5), MIN(Value6) FROM MultiMeter WHERE (DeviceRowID='%q' AND Date>='%q')",
	   sd[0].c_str(), szDate);
...
std::vector<std::string> sd2 = result2[0];
unsigned long long total_min_usage_1 = std::strtoull(sd2[0].c_str(), nullptr, 10);
unsigned long long total_min_deliv_1 = std::strtoull(sd2[1].c_str(), nullptr, 10);
unsigned long long total_min_usage_2 = std::strtoull(sd2[2].c_str(), nullptr, 10);
unsigned long long total_min_deliv_2 = std::strtoull(sd2[3].c_str(), nullptr, 10);
unsigned long long total_real_usage, total_real_deliv;

total_real_usage = powerusage - (total_min_usage_1 + total_min_usage_2);
total_real_deliv = powerdeliv - (total_min_deliv_1 + total_min_deliv_2);
```

In my case, the last line of code will be: total_real_deliv = 571 - (571 + 2) = -2

So basically, the issue is in those first two lines of code, where a check is being done whether the counter values are less than 10. I can imagine this check is included to prevent false readings or very low readings and show them as 0. However, I think this check and correction should then also be applied on the values which are fetched from the MultiMeter database table.

**Comment by gizmocuz** [comment]
Have a look in the database. (Use a tool like SQLite Expert Personal)
First in the DeviceStatus table, find your device, remember the ID and also the svalue (svalue is a combination of different values, current meter T1/T2, return meter T1/T2 and usage/delivery)
Next go to the MultiMeter and MultiMeter_Calendar, do a filter on the DeviceRowID, set it to the above ID
Now you see all values of the P1 meter... I think your problem lies in the multimeter_calendar and with one of the 4 counters

**Comment by gizmocuz** [comment]
@fwieringen , thanks for looking into this. The only thing i don't get is, your counter is set to 2, this will be set to 0 ?
Could you please provide the sValue of your P1 meter, and all fields of the first record of MultiMeter, I think with this we should be able to reproduce it

**Comment by gizmocuz** [comment]
(And a counter value of 2 is maybe an error in the P1 meter itself, still it is so low , we might need to silence that)

**Comment by fwieringen** [comment]
Correct assumption, there was a weird spike on February 3. This caused the counter to be updated to 2. It has been reporting 2 ever since. 

```
SELECT * FROM MultiMeter WHERE DeviceRowID = 1 AND Value4>0;
1|15835079|571|0|7244|11920537|2|2022-02-03 09:00:00

SELECT * FROM DeviceStatus WHERE ID = 1;
1|2|1|1|Energie|1|250|1|2|1|12|255|0|15890274;11949330;571;2;1638;0|2022-02-06 14:04:23|1|0.0|1.0|0.0|1.0|||0|0|0|||
```

**Comment by fwieringen** [comment]
> (And a counter value of 2 is maybe an error in the P1 meter itself, still it is so low , we might need to silence that)

Or change line 10369 from the code to:
```
total_real_deliv = powerdeliv - (total_min_deliv_1 + total_min_deliv_2);
total_real_deliv = total_real_deliv < 0 ? 0 ; total_real_deliv;
```

**Comment by gizmocuz** [comment]
Yes but I would like to know why it is negative first... going to do some testing with a dummy P1 sensor now

**Comment by gizmocuz** [comment]
With your values from the DeviceState and MultiMeter I created a Dummy P1 Meter, my results look like this

![image](https://user-images.githubusercontent.com/2265299/152682891-37ab5e94-b88b-4423-963f-59a0c45eddda.png)

Why did you issue AND Value4>0 ?
Could you send me the value at 00:00:00 (the first value of the day), maybe copy/paste it, or select/query the row number

**Comment by fwieringen** [comment]
> Yes but I would like to know why it is negative first... going to do some testing with a dummy P1 sensor now

It's negative because:

- there's a correction applied when the actual meter value is < 10. It's then set to 0 (line 10317 and 10318);
- there's **no** correction applied when the start of day meter value is queried from the database (line 10362+)
- substracting an uncorrected value from a corrected value could result in a negative value. In my case this is 571 + 0 (corrected) - (571 + 2) = -2

Happy to hear the test results with the dummy P1.

Cheers, Friso

**Comment by gizmocuz** [comment]
Okey now I see, there is a correction applied when the value < 2 (not 10 ... but we could set it to 10 maybe later)
Still the first row of the day for this meter would be good to simulate it

**Comment by gizmocuz** [comment]
My powerdelivered is 571, could be your SQL columns are in different order, maybe a screenshot of the first row will be better

**Comment by fwieringen** [comment]
> 

The AND Value4>0 is to identify the moment of the weird meter value.

Here's the query result of the query in the code:

```
SELECT MIN(Value1), MIN(Value2), MIN(Value5), MIN(Value6) FROM MultiMeter WHERE (DeviceRowID='1' AND Date>='2022-02-06'
15879556|571|11494330|2
```

**Comment by fwieringen** [comment]
> 7244

There are two corrections. One on the individual counter level (10) and one on the summarized counters (2):

```
powerdeliv1 = (powerdeliv1 < 10) ? 0 : powerdeliv1;
powerdeliv2 = (powerdeliv2 < 10) ? 0 : powerdeliv2;

unsigned long long powerusage = powerusage1 + powerusage2;
unsigned long long powerdeliv = powerdeliv1 + powerdeliv2;
if (powerdeliv < 2)
powerdeliv = 0;
```

**Comment by gizmocuz** [comment]
Thanks, with your last row data I can simulate it
Still after correcting it... it will end up like this

![image](https://user-images.githubusercontent.com/2265299/152683431-5f5b1e42-a2ad-4382-b944-962fc21fc4e7.png)

There are two tariffs T1 and T2, for both Usage and Delivery ... 

One counter (T1) is set to 2, the other T2 to 571 ... Do you have solar panels ?

Edit: On your actual meter you should be able to see those values as well I suppose

**Comment by gizmocuz** [comment]
If you don't deliver back, the value '10' we should probably change to 1000
This also has to be done when generating the report overviews else you will get two delivery columns there as well

**Comment by fwieringen** [comment]
> Thanks, with your last row data I can simulate it Still after correcting it... it will end up like this
> 
> ![image](https://user-images.githubusercontent.com/2265299/152683431-5f5b1e42-a2ad-4382-b944-962fc21fc4e7.png)
> 
> There are two tariffs T1 and T2, for both Usage and Delivery ...
> 
> One counter (T1) is set to 2, the other T2 to 571 ... Do you have solar panels ?
> 
> Edit: On your actual meter you should be able to see those values as well I suppose

Actually T1 = 571 and T2 = 2. This 2 occured since Feb 03, when we had a weird spike around 9 AM. This is High Tariff time.

The values on the meter are 0, because the meter only shows kWh and no decimals.

Correcting it to 1000 solves it for the short term, but as soon as someone has these minor spikes once in a while, the counter will exceed 1000 at a moment in time and you will have the same issue ;-)

We're in a temp home at the moment without solar panels. I've been using my RPi with Domoticz since two weeks now and have no idea how the 571 on T1 is created in the past. Probably similar power spikes/surges.

**Comment by gizmocuz** [comment]
Yes I always swap the two... 
Pushed a fix, and building new beta versions....
Still I suppose you don't want to display the return data of this meter... maybe this should be made a option some day

**Comment by fwieringen** [comment]
> Yes I always swap the two... Pushed a fix, and building new beta versions.... Still I suppose you don't want to display the return data of this meter... maybe this should be made a option some day

I think that would be a good addition. Thanks for looking into this. I will figure out how to install this beta and will let you know.

**Comment by gizmocuz** [comment]
Should be fixed in beta 14156

**Comment by uffer** [comment]
Confirmed:

![afbeelding](https://user-images.githubusercontent.com/18116064/152696598-1bc332e3-920f-4206-af10-2eaacf36ae60.png)

Thanx for solving this issue!

**Comment by stafil** [comment]
Hi,

I have same problem.

![image](https://user-images.githubusercontent.com/4068371/208089814-00f54e58-f6e1-441b-ac38-a89b3112a6cd.png)

I send this command: 

/json.htm?type=command&param=udevice&idx=492&nvalue=0&svalue=10;11;12;0;-14;0

[DOMOTICZ2022.2 (build 14854)

thx

**Comment by gizmocuz** [comment]
@stafil , I don't think so, and the P1 meter does not work with negative values
Best you open the database in a editor (SQLite Expert Personal edition, for example) and correct the issue
